src/main/java/com/oltpbenchmark/WorkloadState.java

```diff
@@ -71,11 +71,17 @@ public void addToQueue(int amount, boolean resetQueues) {
                 return;
             } else {
                 // Add the specified number of procedures to the end of the queue.
-                for (int i = 0; i < amount; ++i) {
+                for (int i = 0; i < amount && workQueue.size() <= RATE_QUEUE_LIMIT; ++i) {
                     workQueue.add(new SubmittedProcedure(currentPhase.chooseTransaction()));
                 }
             }
 
+            // The following approach has 2 issues. 
+			// 1. If the rate is very high, it unnecessarily enqueues and dequeues for the excessive amount 
+			//    every time the work is added to the queue.
+			// 2. If the rate is very high, the memory overflow occurs. Even if we increase the Java heap size,
+            //    if the rate is also raised, the memory error will occur again. 
+			//    We can elminitate this memory error entirely with a simple fix of the algorithm.
```

Comment:
Fix formatting (use spaces, not tabs).
This comment doesn't make sense here. Move it up above to the `for` loop.